track_and_trace/asset_client/src/views/add_asset_form.js

```diff
@@ -134,81 +107,42 @@ const AddAssetForm = {
   }
 }
 
-/**
- * Update the reporter's values after a change occurs in the name of the
- * reporter at the given reporterIndex. If it is empty, and not the only
- * reporter in the list, remove it.  If it is not empty and the last item
- * in the list, add a new, empty reporter to the end of the list.
- */
-const _updateReporters = (vnode, reporterIndex) => {
-  let reporterInfo = vnode.state.reporters[reporterIndex]
-  let lastIdx = vnode.state.reporters.length - 1
-  if (!reporterInfo.reporterKey && reporterIndex !== lastIdx) {
-    vnode.state.reporters.splice(reporterIndex, 1)
-  } else if (reporterInfo.reporterKey && reporterIndex === lastIdx) {
-    vnode.state.reporters.push({
-      reporterKey: '',
-      properties: []
-    })
-  }
+const _clearForm = (state) => {
+  location.reload()
 }
 
 /**
  * Handle the form submission.
  *
  * Extract the appropriate values to pass to the create record transaction.
  */
-// const _handleSubmit = (signingKey, state) => {
-//   const properties = [{
-//     name: 'type',
-//     stringValue: state.type,
-//     dataType: payloads.createRecord.enum.STRING
-//   }]
-
-//   if (state.subtype) {
-//     properties.push({
-//       name: 'subtype',
-//       stringValue: state.subtype,
-//       dataType: payloads.createRecord.enum.STRING
-//     })
-//   }
-
-//   if (state.weight) {
-//     properties.push({
-//       name: 'weight',
-//       numberValue: parsing.toInt(state.weight),
-//       dataType: payloads.createRecord.enum.NUMBER
-//     })
-//   }
-
-//   if (state.latitude && state.longitude) {
-//     properties.push({
-//       name: 'location',
-//       locationValue: {
-//         latitude: parsing.toInt(state.latitude),
-//         longitude: parsing.toInt(state.longitude)
-//       },
-//       dataType: payloads.createRecord.enum.LOCATION
-//     })
-//   }
-
-//   const recordPayload = payloads.createRecord({
-//     recordId: state.serialNumber,
-//     recordType: 'asset',
-//     properties
-//   })
-
-//   const reporterPayloads = state.reporters
-//     .filter((reporter) => !!reporter.reporterKey)
-//     .map((reporter) => payloads.createProposal({
-//       recordId: state.serialNumber,
-//       receivingAgent: reporter.reporterKey,
-//       role: payloads.createProposal.enum.REPORTER,
-//       properties: reporter.properties
-//     }))
-
-//   transactions.submit([recordPayload].concat(reporterPayloads), true)
-//     .then(() => m.route.set(`/assets/${state.serialNumber}`))
-// }
+const _handleSubmit = (state) => {
+  const properties = [{
+    name: 'serialNumber',
+    dataType: PropertyDefinition.DataType.STRING,
+    stringValue: state.serialNumber
+  },
+  {
+    name: 'type',
+    stringValue: state.type,
+    dataType: PropertyDefinition.DataType.STRING
+  },
+  {
+    name: 'weight',
+    numberValue: (state.weight ? parsing.toInt(state.weight) : 0),
```

Comment:
It should not be possible to submit the transaction if the weight or location is not set. It is possible for an asset to actually validly have a weight of 0 or a location of 0,0, so setting it to zero is not correct.

Reply:
updated